tswow-core/Public/global.d.ts

```diff
@@ -2967,6 +2967,20 @@ declare interface TSCreature extends TSUnit {
      */
     AttackStart(target : TSUnit) : void
 
+    /**
+     * Sets how a [Creature] responds to being attacked.
+     *
+     * @param uint8 state : the state to set. 0 = Passive, 1, = Defensive, 2 = Aggressive
+     */
+    SetReactState(state: uint8) : void
+
+    /**
+     * Gets how a [Creature] responds to being attacked.
+     *
+     * @return uint8 0 = Passive, 1, = Defensive, 2 = Aggressive
+     */
+    GetReactState() : uint8
+
```

Comment:
We want to return/accept proper enums here instead of raw uint8s.

If you look at the top of this file, you'll see that we declare a bunch of enums, and you'll see a few of them are completely empty. The two ways to add enums here are:

1. Automatically build them from trinitycore, these are the empty ones. You do it by finding a suitable enum in a trinitycore source file and writing it as `declare const enum MyEnumName {} /** HeaderFile.h:EnumName */`. This should always be preferred where possible, which it should be here (ReactStates are in `UnitDefines.h`). 

2. Writing the values manually, that's the non-empty ones. This should only be done when there is no corresponding enum in trinitycore, or the existing gives errors if we try to build it. You do this by writing it as `declare const MyEnumName /**@realType:uintType*/ { <then any contents in here> }`

You can test that this works by doing `build trinitycore-relwithdebinfo --no-compile`, it will give an error if any enum cannot be built.

Reply:
I've done this but the output in global.d.ts doesn't match my expectations.

```ts
declare const enum ReactStates {} /** UnitDefines.h:ReactStates */
```

Output

```ts
declare const enum ReactStates /**@realType:uint8*/ {
    PAS    = 0,
    DEFEN  = 1,
    AGGRES = 2
}
```

Reply:
There was a typo in the generator code, try merging 448d0ffda1dda6dec8e9a188999cc7ef94f484a2

Reply:
![image](https://user-images.githubusercontent.com/7526918/150877413-459497ce-3a42-4e44-bd46-ba773b49e3b7.png)

Resolved.